src/pages/Book.jsx

```diff
@@ -1,15 +1,92 @@
-import { Link, useLocation } from "react-router-dom";
+import { Link, useLocation, useNavigate, useParams } from "react-router-dom";
+import axios from "axios";
 import "../styles/Book.css";
+import { useEffect, useState } from "react";
 
 function Book() {
+  const { id } = useParams();
   const location = useLocation();
   const { book } = location.state;
+  const [chapters, setChapters] = useState([]);
+  const navigate = useNavigate();
+
+  useEffect(() => {
+    const getListOfChapters = async () => {
+      try {
+        const res = await axios.post(
+          "https://api-preprod.lelivrescolaire.fr/graph",
+          {
+            query:
+              "query chapters($bookId:Int){viewer{chapters(bookIds:[$bookId]){hits{id title url valid}}}}",
+            variables: {
+              bookId: parseInt(id),
+            },
+          },
+          {
+            headers: {
+              "Content-Type": "application/json; charset=utf-8",
```

Comment:
You can define this header for the axios instance instead of having to redefine it every time.